libs/community/langchain_community/vectorstores/azure_cosmos_db_no_sql.py

```diff
@@ -274,15 +274,15 @@ def _similarity_search_with_score(
             query += "TOP @limit "
 
         query += (
-            "c.id, c.{}, c.text, c.metadata, "
-            "VectorDistance(c.@embeddingKey, @embeddings) AS SimilarityScore FROM c"
+            "c.id, c[@embeddingKey], c.text, c.metadata, "
+            "VectorDistance(c[@embeddingKey], @embeddings) AS SimilarityScore FROM c"
         )
 
         # Add where_clause if specified
         if pre_filter is not None and pre_filter.get("where_clause") is not None:
             query += " {}".format(pre_filter["where_clause"])
 
-        query += " ORDER BY VectorDistance(c.@embeddingKey, @embeddings)"
+        query += " ORDER BY VectorDistance(c[@embeddingKey], @embeddings)"
```

Comment:
I'm not very familiar with Azure CDB NoSQL but is `c.@embeddingKey` equivalent to `c[@embeddingKey]`? Why make this change?

Reply:
As far as I know, the dot syntax is not supported for variable expressions. You are therefore forced to use brackets if you want to protect yourself against injections. CDB NoSQL seems quite restrictive in its capabilities here. But I do not have greater proficiency in it either, so there may be smarter ways about it.